web-app/js/portal/filter/ui/FilterGroupPanel.js

```diff
@@ -63,7 +63,10 @@ Portal.filter.ui.FilterGroupPanel = Ext.extend(Portal.filter.ui.GroupPanel, {
     _handleGetFeatureRequestResults: function(results) {
         if (results.status == 200) {
             var res = Ext.util.JSON.decode(results.responseText);
-            this.dataCollection.totalFilteredFeatures = (res && res.totalFeatures >= 0) ? res.totalFeatures: undefined;
+            if (res && res.totalFeatures == "unknown") {
+                res.totalFeatures = res.features.length;
```

Comment:
We've just lost the returned totalFeatures.  Why not use a temporary variable instead of modifying the response.